### PR TITLE
Take keywords and generic name into account for relevancy

### PR DESCRIPTION
**Pull request by Gaarco**
## Description
Attempt to address #347 #421 

Performs fuzzy search on name, generic name and keywords and uses the best result as score.
Also adjusted the threshold to include more relevant applications since some were previously excluded even though the app name included the search keyword.

### Submitter Checklist

- [x] Squashed commits with `git rebase -i` (if needed)
- [x] Built budgie-desktop and verified that the patch worked (if needed)

### CODE REVIEW

**Review by EbonJaeger — COMMENTED** [review]
This looks pretty good. Thanks! Do you have any examples of testing, like "application A used to be lower/not shown when expected, and now it is"? Just so we have something to go off of.

File: src/appindexer/RelevancyService.vala [hunk 1]
```diff
@@ -159,6 +162,28 @@ namespace Budgie {
...
+				if (e == null) {
+					continue;
+				}
+
+				string target = searchable_string(e);
+				int tmp_score = Fuzzer.get_fuzzy_score(target, term, 1);
+
+				if (tmp_score >= 0 && (tmp_score < score || score == -1)) {
+					score = tmp_score;
+				}
```
[the remark this answers is not in the document]
Comment (Gaarco) [reply src/appindexer/RelevancyService.vala:181]: Maybe I misunderstood, but isn't that how the algorithm works? The lower the score the higher the relevancy

[the remark this answers is not in the document]
Comment (EbonJaeger) [reply src/appindexer/RelevancyService.vala:181]: :facepalm: Yes, of course. Derp. I don't know what I was thinking. Nevermind. That's what I get for making lower == better.

**Comment by Gaarco** [comment]
Searching for "GIMP" didn't show anything before, now it displays the application.
When searching for "settings," it used to show only Budgie Desktop Settings, it now also displays the Budgie Control Center, PulseAudio Volume Control and others.

**Review by EbonJaeger — APPROVED** [review]
Looks good to me, and it works really well for searching for generic terms like "browser", "editor", or "terminal"! Kicking over to Josh for final review. :)

**Review by JoshStrobl — APPROVED** [review]
LGTM, thanks!

**Pull request merged**